pandas/core/indexes/multi.py

```diff
@@ -508,7 +508,9 @@ def from_product(cls, iterables, sortorder=None, names=None):
             Level of sortedness (must be lexicographically sorted by that
             level).
         names : list / sequence of str, optional
-            Names for the levels in the index.
+            Names for the levels in the index. If not provided, these
+            will be inferred from iterables if the iterable has a
```

Comment:
This should rather be "if the elements of \`iterables\` have a `name` attribute"?

Reply:
This is updated now.